node: forward public REST requests to public gRPC

tbjump opened the pull request.

tbjump commented on node/cmd/guardiand/node.go:
```diff
@@ -497,6 +497,9 @@ func runNode(cmd *cobra.Command, args []string) {
 	if *adminSocketPath == "" {
 		logger.Fatal("Please specify --adminSocket")
 	}
+	if (*publicRPC == "") != (*publicWeb == "") {
+		logger.Fatal("Please either specify both --publicRPC and --publicWeb or leave both empty")
+	}
```
maybe. All the integration tests are passing :)

tbjump replied:
If node operators only want to make one of the two public, they could still specify the loopback interface or block it through firewall. Curious what @SEJeff thinks.

tbjump replied:
addressed in latest commit (see discussion)

evan-gray commented:
Concur with the concerns. Also just noting that the long term goal is to make the public endpoints unnecessary via https://github.com/wormhole-foundation/wormhole-explorer/issues/4

tbjump commented:
@leoluk @evan-gray I addressed the concerns by adding an additional unix socket just for the public gRPC. That way, guardians wouldn't run into firewall issues and the performance concern should also be addressed. This way, a guardian can also decide to have --publicRPC or --publicWeb but doesn't need to have both. 
What do you think?

jynnantonix reviewed (dismissed):
Approving because the code change itself looks good to me but let's make sure @leoluk and @evan-gray have no more concerns before merging.

tbjump commented:
Required tests are passing now

hendrikhofstadt commented:
Why is the socket listener required for the tcp listener to work? I don't think I'm following the rationale behind this change

tbjump commented:
> Why is the socket listener required for the tcp listener to work? I don't think I'm following the rationale behind this change

The json-web listener forwards to grpc socket. We could forward the json-web to grpc-tcp, as I initially proposed, but then the json-web depends on the grpc-tcp (which it didn't before) and it may create firewall and performance issues according to @leoluk. Therefore in the latest proposal we create a separate socket for public requests.
